src/main/java/uk/gov/laa/ccms/caab/controller/submission/ClientSubmissionsInProgressController.java

```diff
@@ -57,15 +59,44 @@ public String submissionInProgress(
       return "redirect:/submissions/client-create/confirmed";
     }
 
-    return viewIncludingPollCount(session);
+    return viewIncludingPollCount(session, SUBMISSION_CREATE_CLIENT);
   }
 
-  private String viewIncludingPollCount(HttpSession session) {
+  /**
+   * Handles the GET request for the client update submission in progress screen.
+   *
+   * @return The view name for the submission in progress.
+   */
+  @GetMapping("/submissions/client-update")
+  public String clientUpdateSubmission(
+      @SessionAttribute(SUBMISSION_TRANSACTION_ID) String transactionId,
+      @SessionAttribute(USER_DETAILS) UserDetail user,
+      HttpSession session, Model model) {
+
+    model.addAttribute("submissionType", SUBMISSION_UPDATE_CLIENT);
+
+    ClientStatus clientStatus = clientService.getClientStatus(
+        transactionId,
+        user.getLoginId(),
+        user.getUserType()).block();
+
+    if (clientStatus != null && StringUtils.hasText(clientStatus.getClientReferenceNumber())) {
+      //Do some session tidy up
+      session.removeAttribute(SUBMISSION_TRANSACTION_ID);
+      session.removeAttribute(CLIENT_FLOW_FORM_DATA);
+
+      return "redirect:/submissions/client-update/confirmed";
+    }
+
+    return viewIncludingPollCount(session, SUBMISSION_UPDATE_CLIENT);
+  }
+
+  private String viewIncludingPollCount(HttpSession session, String submissionType) {
     int submissionPollCount = 0;
     if (session.getAttribute("submissionPollCount") != null) {
       submissionPollCount = (int) session.getAttribute("submissionPollCount");
       if (submissionPollCount >= 6) {
-        return "redirect:/submissions/client-create/failed";
+        return String.format("redirect:/submissions/%s/failed", submissionType);
```

Comment:
hardcoded max poll count to be externalised, as discussed

Reply:
done